applications/ShallowWaterApplication/custom_processes/write_from_sw_at_interface_process.cpp

```diff
@@ -0,0 +1,223 @@
+    mDirection = -mrVolumeModelPart.GetProcessInfo()[GRAVITY];
+    mDirection /= norm_2(mDirection);
+    mPrintVelocityProfile = ThisParameters["print_velocity_profile"].GetBool();
+
+    if (!mStoreHistorical) {
+        VariableUtils().SetNonHistoricalVariableToZero(MOMENTUM, mrInterfaceModelPart.Nodes());
+        VariableUtils().SetNonHistoricalVariableToZero(VELOCITY, mrInterfaceModelPart.Nodes());
+        VariableUtils().SetNonHistoricalVariableToZero(HEIGHT, mrInterfaceModelPart.Nodes());
+        VariableUtils().SetNonHistoricalVariableToZero(VERTICAL_VELOCITY, mrInterfaceModelPart.Nodes());
+    }
+
+    // if (mExtrapolateBoundaries) {
+    //     FindBoundaryNeighbors();
+    // }
+}
+
+template<std::size_t TDim>
+void WriteFromSwAtInterfaceProcess<TDim>::Execute()
+{
+    
+    BinBasedFastPointLocator<TDim> locator(mrVolumeModelPart);
+    locator.UpdateSearchDatabase();
+
+    struct locator_tls {
+        Vector N;
+        typename BinBasedFastPointLocator<TDim>::ResultContainerType results;
+        locator_tls(const int max_results = 10000) {
+            N.resize(TDim+1);
+            results.resize(max_results);
+        }
+    };
+    
+    block_for_each(mrInterfaceModelPart.Nodes(), locator_tls(), [&](NodeType& rNode, locator_tls& rTLS){
+        ReadAndSetValues(rNode, locator, rTLS.results);
+    });
+
+    if (mExtrapolateBoundaries) {
+        CopyValues(*mpFirstBoundaryNeighbor, *mpFirstBoundaryNode);
+        CopyValues(*mpSecondBoundaryNeighbor, *mpSecondBoundaryNode);
+    }
+}
+
+// template<std::size_t TDim>
+// void WriteFromSwAtInterfaceProcess<TDim>::GetBoundingVolumeLimits(double& rMin, double& rMax)
+// {
+//     using MultipleReduction = CombinedReduction<MinReduction<double>,MaxReduction<double>>; 
+
+//     std::tie(rMin, rMax) = block_for_each<MultipleReduction>(mrVolumeModelPart.Nodes(), [&](NodeType& node){
+//         const double distance = inner_prod(mDirection, node);
+//         return std::make_tuple(distance, distance);
+//     });
+// }
+
+template<std::size_t TDim>
+void WriteFromSwAtInterfaceProcess<TDim>::ReadAndSetValues(
+    NodeType& rNode,
+    BinBasedFastPointLocator<TDim>& rLocator,
+    typename BinBasedFastPointLocator<TDim>::ResultContainerType& rResults)
+{
+    
+    // initialize the integration variables
+    Element::Pointer p_elem;
+    array_1d<double,3> velocity = ZeroVector(3);
+    array_1d<double,3> momentum = ZeroVector(3);
+    double height = 0.0;
+    double vertical_velocity = 0.0;
+    double topography = 0.0;
+    
+    velocity = rNode.FastGetSolutionStepValue(VELOCITY);
+    momentum = rNode.FastGetSolutionStepValue(MOMENTUM);
+    height = rNode.FastGetSolutionStepValue(HEIGHT);
+    vertical_velocity = rNode.FastGetSolutionStepValue(VERTICAL_VELOCITY);
+    topography = rNode.FastGetSolutionStepValue(TOPOGRAPHY);
+
+    SetValue(rNode, MOMENTUM, momentum);
+    SetValue(rNode, VELOCITY, velocity);
```

Comment:
does it make sense the do SetValue and not `SetSolutionStepValue`. I do not understand

Reply:
This is due to some issue in the HDF5Application, for which all variables are considered as NonHistorical.